### PR TITLE
Static analysis #2

### PR DESCRIPTION
**Pull request by dkalinowski**
- adding check for errors from status details getter
- adding missing breaks in deserialization switch
- fixing leak in ovms start

### CODE REVIEW

**Review by rasapala — APPROVED** [review]
LGTM

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 1]
```diff
@@ -36,14 +36,24 @@ namespace mediapipe {
 
 using std::endl;
 
+
+#define THROW_IF_FATAL_ERR(C_API_CALL)                                          \
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc]:
```suggestion
#define THROW_IF_RECURSIVE_ERR(C_API_CALL)                                          \
```

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc [hunk 2]
```diff
@@ -44,14 +44,21 @@ namespace mediapipe {
 using std::endl;
 
 namespace {
+
+#define ASSERT_FATAL_ERR(C_API_CALL)    \
```
Comment (atobiszei) [inline mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: RECURSIVE

**Review by atobiszei — APPROVED** [review]

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc (hunk 1 above)
Comment (dkalinowski) [reply mediapipe/calculators/ovms/modelapiovmsadapter.cc]: done CIRCULAR

File: mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc (hunk 2 above)
Comment (dkalinowski) [reply mediapipe/calculators/ovms/modelapiovmsinferencecalculator.cc]: done CIRCULAR

**Pull request merged**